### PR TITLE
Added -hud ZAP command line option

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            argClasses[i] = arg.getClass();
+            i++;
+        }
+        Method method = obj.getClass().getMethod(methodName, argClasses);
+        return method.invoke(obj, args);
+    }
+
+    private void launchBrowser() {
+        new Thread(
+                        new Runnable() {
```
Comment (thc202) [inline src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java]: Could use a lambda.

File: src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                                        getExtHUD().setHudCmdlineOptionUsed(true);
+                                        getExtHUD().getHudParam().getConfig().save();
+                                    }
+                                    // Object wd =
+                                    callMethodByReflection(
+                                            getExtSelenium(), "getProxiedBrowserByName", "Firefox");
+
+                                    // callMethodByReflection(wd, "get", "https://zaproxy.org");
+                                } catch (Exception e1) {
+                                    if (!View.isInitialised()) {
```
Comment (thc202) [inline src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java]: Log always? Should ZAP exit if not able to start the browser when in daemon?

File: src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                        },
+                        "ZAP-BrowserLauncher")
+                .start();
+    }
+
+    @Override
+    public void execute(CommandLineArgument[] args) {
+        if (arguments[ARG_BROWSER_IDX].isEnabled()) {
+            this.launchBrowser();
+        } else {
```
Comment (thc202) [inline src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java]: The else can be removed.

File: src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java (hunk 2 above)
Comment (psiinon) [reply src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java]: I think exiting is a bit too harsh in this case

**Comment by psiinon** [comment]
Think that should address all of the comments

**Comment by thc202** [comment]
Thanks!

**Pull request merged**